Sources/ArcGISToolkit/Components/Offline/PreplannedMapModel.swift

```diff
@@ -141,7 +141,14 @@ class PreplannedMapModel: ObservableObject, Identifiable {
         } catch {
             status = .mmpkLoadFailure(error)
         }
-        return mobileMapPackage.maps.first
+        if let map = mobileMapPackage.maps.first {
+            // Add the preplanned map area ID as a tag to differentiate between
+            // local map areas.
+            map.item!.addTag(preplannedMapAreaID.rawValue)
+            return map
+        } else {
+            return nil
+        }
```

Comment:
When these changes were added, there were some difficulties to determine if a `Map` object is created from a certain mmpk or not. A few things have been tried:

- Use the local item's various IDs. The item ID is different from its portal item ID; the `originalPortalItemID` is the portal item ID of the parent webmap, not the preplanned map areas'
- Use the local item's `fileURL`. Because a `Map` created from a mmpk is sometimes compressed, this property can be `nil`
- Use the object itself. This would require loading and creating a `Map` object from each mmpk, which is not efficient and asynchronous
- Use other metadata. Title, description, and snippet aren't guaranteed to be unique for each preplanned map area

Eventually I decided to add the preplanned map area's portal item ID as a tag to the local item for differentiating each map object, so it is easy to tell which mmpk it was created from.

Reply:
Because a `mobileMapPackage` needs to be loaded in order to access the maps in it. In current implementation, we only load the packages "on-demand" (when the open button is tapped) so that only 1 package is loaded at a time, to save resources. Thus, when the comparison Boolean is computed, the `model.map` is `nil`.

Reply:
After a long winding detour, it turned out that the local item's title is unique.